backend-app/controllers/auth_controllers/auth_controller.ts

```diff
@@ -12,84 +11,97 @@ import {
 import AuthUtils from '@utils/authorization/auth_utils';
 import searchCookies from '@utils/searchCookie';
 import User from '@models/user/user_model';
-import { Request, Response, NextFunction } from 'express';
-import { IUser } from '@root/interfaces/models/i_user';
+import {
+    Request,
+    Res,
+    TsoaResponse,
+    Controller,
+    Get,
+    Post,
+    Tags,
+    Query,
+    Body,
+    Route,
+} from 'tsoa';
+import { IReq } from '@root/interfaces/vendors';
 
 const generateActivationKey = async () => {
     const randomBytesPromiseified = promisify(require('crypto').randomBytes);
     const activationKey = (await randomBytesPromiseified(32)).toString('hex');
     return activationKey;
 };
 
-export const githubHandler = async (
-    req: Request,
-    res: Response,
-    next: NextFunction
-): Promise<void> => {
-    try {
-        const Roles = await Role.getRoles();
-        // check if user role exists
-        if (!Roles.USER)
-            throw new AppError(
-                500,
-                'User role does not exist. Please contact the admin.'
-            );
-        const { code } = req.query as {
-            code: string;
-        };
+@Route('/api/auth')
+@Tags('Authentication')
+export class AuthController extends Controller {
+    @Get('github/callback')
```

Comment:
+@Examples()
Is there any reason you are not using  this?

Reply:
What do you mean by not using it

Reply:
You have told me to use this decorator. But you havent used it.

Reply:
Ah , example decorator , did I hhh
It is okay I will use it, I just forgot 🙂

Reply:
what do you think about this :
    buildOperation(controllerName, method, defaultProduces) {
        const swaggerResponses = {};
        method.responses.forEach((res) => {
            if (swaggerResponses[res.name]) {
                swaggerResponses[res.name].description += ` \n - ${res.description}`;
            } else {
                swaggerResponses[res.name] = {
                    description:  '- '+ res.description,
                };
            }
i have modified tsoa package to achive the aproach of multiple response with same status code :
    @Response(400, 'Please provide email and password')
    @Response(400, 'Invalid email or password')
    @Response(400, 'You haven\'t set a password yet. Please login with GitHub and set a password from your profile page.')
it is working but the problem is that any one who installs the project he will need to modify the package !
what do you think ? can you find a better aproach ?